modules/cfe_testcase/src/time_conversion_test.c

```diff
@@ -0,0 +1,150 @@
+    OS_time_t          difference;
+
+    /* Mission Elapsed Time */
+    CFE_TIME_SysTime_t MET;
+    /* MET + SCTF */
+    CFE_TIME_SysTime_t TAI;
+    /* MET + SCTF - Leap Seconds */
+    CFE_TIME_SysTime_t UTC;
+    /* Spacecraft Time */
+    CFE_TIME_SysTime_t SCTime;
+
+    /* Print buffers */
+    char timeBuf1[sizeof("yyyy-ddd-hh:mm:ss.xxxxx_")];
+    char timeBuf2[sizeof("yyyy-ddd-hh:mm:ss.xxxxx_")];
+
+    OS_GetLocalTime(&start);
+
+    /* Get Times */
+    MET = CFE_TIME_GetMET();
+    TAI = CFE_TIME_GetTAI();
+    UTC = CFE_TIME_GetUTC();
+
+    OS_GetLocalTime(&end);
+
+    /* calculate diffrence */
+    difference = OS_TimeSubtract(end, start);
+
+    /* 
+     * Convert - should produce a TAI or UTC at the moment of GetMET() 
+     *			 with a minimal time shift between 
+     */
+    SCTime = CFE_TIME_MET2SCTime(MET);
+
+    /* write Spacecraft Time into second buffer */
+    CFE_TIME_Print(timeBuf2, SCTime);
+
+    /* Check conversion */
+#if (CFE_MISSION_TIME_CFG_DEFAULT_TAI == true)
+    /* SCTime is TAI format */
+
+    /* avoid unused compiler warning */
+    (void)UTC;
+
+    /* write TAI into first buffer */
+    CFE_TIME_Print(timeBuf1, TAI);
+
+    /*
+     * CFE_TIME_Compare not possible because of the 
+     * temporal diffrence between get GetMET and GetTAI 
+     */
+    UtAssert_True(TimeInRange(TAI, SCTime, difference), "TAI (%s) = MET2SCTime (%s)", timeBuf1, timeBuf2);
+
+    /* this would be ok but it would repeat a whats done in CFE_TIME_MET2SCTime */
+    /* UtAssert_True(CFE_TIME_Compare(SCTime, CFE_TIME_Add(MET, STCF))); */
+#else
+    /* SCTime is UTC format */
+
+    /* avoid unused compiler warning */
+    (void)TAI;
+
+    /* write UTC into first buffer */
+    CFE_TIME_Print(timeBuf1, UTC);
+
+    /* 
+     * CFE_TIME_Compare not possible because of the
+     * temporal diffrence between get GetMETand GetUTC 
+     */
+    UtAssert_True(TimeInRange(UTC, SCTime, difference), "UTC (%s) = MET2SCTime (%s)", timeBuf1, timeBuf2);
+#endif
+}
+
+void TestConvertSubSeconds2MicroSeconds(void)
+{
+    UtPrintf("Testing: CFE_TIME_Sub2MicroSecs");
+
+    /* predefined amount of sub-seconds */
+    uint32 SUB = 400;
+    /* correct micro-seconds equal to the predefined sub-seconds */
+    uint32 MS = 93;
+    uint32 Sub2Micro;
+
+    /* run Sub2MicroSecs with predefined amount of sub-seconds and save result */
+    Sub2Micro = CFE_TIME_Sub2MicroSecs(SUB);
+
+    UtAssert_True(MS == Sub2Micro, "sub-seconds (%u) = Sub2MicroSecs (%u)", SUB, Sub2Micro);
+}
+
+void TestConvertMicroSeconds2SubSeconds(void)
+{
+    UtPrintf("Testing: CFE_TIME_Micro2SubSecs");
+
+    /* micro-seconds */
+    uint32 MS = 64512;
+    /* sub-seconds */
+    uint32 SUB = 2770770;
+    uint32 Micro2Sub;
+
+    /* convert */
+    Micro2Sub = CFE_TIME_Micro2SubSecs(MS);
+
+    /* get sub-seconds form TimeMS and compare with SUB */
+    UtAssert_True(Micro2Sub == SUB, "micro-seconds (%u) = Micro2SubSecs (%u)", MS, Micro2Sub);
```

Comment:
Still need a second test here for when Microseconds is great then 1,000,000

Reply:
Ok, I made it now with >= 1000000 so > 999999, because this would correspond to the implementation of CFE_TIME_Micro2SubSecs.

Reply:
Short question: could i use the <limits.h> header in this project?